### PR TITLE
ci: run kwil-js test

### PR DESCRIPTION
**Pull request by Yaiba**
This pr will run kwil-js once any PR merged into main. Also changed kwil-test-reuse, so that a kwil-js can use  branch/tag/sha input

### CODE REVIEW

**Comment by Yaiba** [comment]
Temporary run the test in CI workflow, so we can test it out without merge into main.  
If the test passed for kwild, then we can move it into kgw-test-reuse @KwilLuke

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -168,7 +168,7 @@ jobs:
           cat docker-compose.yml
           echo "run kwild in docker"
           docker compose up -d
-          chain_id=`curl -s http://localhost:8080/api/v1/chain_info | jq .chain_id`
+          chain_id=`curl -s http://localhost:8080/api/v1/chain_info | jq -r .chain_id`
```
Comment (Yaiba) [inline .github/workflows/ci.yaml:171]: Maybe sleep for 3 seconds, kwil-db could be ramping up and not ready

Comment (jchappelow) [reply .github/workflows/ci.yaml:171]: Clone and build kwil-js while you wait :)

**Comment by Yaiba** [comment]
I moved the test to 'reuse-kgw-test' workflow, with kwild & kgw tests.

@KwilLuke  To test this branch, using the manual-test-kgw like below 
Screenshot 2024-04-05 at 2 31 33 PM

**Comment by KwilLuke** [comment]
Okay this should be good to go.

**Comment by Yaiba** [comment]
> Okay this should be good to go.

cool. I'll rebase and squash

**Review by KwilLuke — APPROVED** [review]

File: .github/workflows/kgw-test-reuse.yaml [hunk 2]
```diff
@@ -165,6 +166,46 @@ jobs:
           sed -i "s/\${UID}:\${GID}/${testUserID}:${testGroupID}/g" test/integration/docker-compose.override.yml
           KIT_LOG_LEVEL=warn go test -count=1 -timeout 0 ./test/integration -run ^TestKGW$ -v
 
+      - name: Run kwil-js test against kwild & kgw
+        run: |
+          cd deployments/compose/kwil
+          echo "run kwild in docker"
+          docker compose up -d
```
Comment (Yaiba) [inline .github/workflows/kgw-test-reuse.yaml:173]: Suprisingly this doesn't require the 'docker-compose.overwrite' to bypass the permission issue like we did in 'CI' workflow, but it works.

**Pull request merged**